### ISSUE TITLE
[BUG] [QE] Panic when running integration test 

### ISSUE THREAD

**Issue by vyasgun** [crc-org/crc#4335]
### General information

  * OS: macOS
  * Hypervisor: vfkit
  * Did you run `crc setup` before starting it (Yes/No)? Yes
  * Running CRC on: Laptop 

## CRC version
```bash
# Put `crc version` output here
CRC version: 24.08.27+e222fc
OpenShift version: 4.16.7
MicroShift version: 4.16.7
```
  
## Host Operating System
```bash
ProductName:		macOS
ProductVersion:		14.4
BuildVersion:		23E214
```

### Steps to reproduce
This can be reproduced by running integration test on the local machine. For example:
```bash
make GINKGO_OPTS="--ginkgo.label-filter='vm-resize'" BUNDLE_PATH="--bundle-path=/Users/gvyas/.crc/cache/crc_vfkit_4.16.7_arm64.crcbundle" PULL_SECRET_PATH="--pull-secret-path=/Users/gvyas/Downloads/pull-secret.txt" integration
```

### Expected

No panic
### Actual
```
Ran 20 of 36 Specs in 703.655 seconds
SUCCESS! -- 20 Passed | 0 Failed | 0 Pending | 16 Skipped
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x1010efb4c]

goroutine 1 [running]:
time.(*Timer).Stop(...)
	/opt/homebrew/Cellar/go/1.23.0/libexec/src/time/sleep.go:114
testing.(*M).stopAlarm(...)
	/opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:2392
testing.(*M).Run(0x140000feaa0)
	/opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:2037 +0x65c
github.com/crc-org/crc/v2/test/integration_test.TestMain(0x140000feaa0)
	/Users/gvyas/work/crc/test/integration/testsuite_test.go:38 +0x90
main.main()
	_testmain.go:47 +0x98
FAIL	github.com/crc-org/crc/v2/test/integration	704.552s
FAIL
make: *** [integration] Error 1
```

### Logs

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
$ crc cleanup
$ crc setup
$ crc start --log-level debug
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

**Comment by vyasgun** [comment]
The root cause of this panic is that there is a race condition due to `flag.Parse` being called concurrently in two places: https://github.com/crc-org/crc/blob/main/test/integration/testsuite_test.go#L36 and https://github.com/crc-org/crc/blob/main/test/extended/util/proxy.go#L75 (The function here is called as a goroutine in the proxy_test.go file).

Another issue that I came across while working on this is that `UserMode` is not returned as default in the current code which causes certain issues in running the tests (in the case of MacOS and Windows). The PR for this will address this as well.

**Comment by anjannath** [comment]
fixed in #4380
